src/main/java/dk/sdu/seb05/semesterprojekt/DomainLayer/DomainController.java

```diff
@@ -0,0 +1,232 @@
+package dk.sdu.seb05.semesterprojekt.DomainLayer;
+
+import dk.sdu.seb05.semesterprojekt.PersistenceLayer.*;
+import dk.sdu.seb05.semesterprojekt.PersistenceLayer.JSONController.JSONController;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class DomainController implements IDomainController {
+    IDataLayer dataLayer = new JSONController();
+    Session session = new Session();
+    Category[] categories = Category.values();
+    List<FunctionType> functionTypes = FunctionType.getAllFunctionTypes();
+    List<IProgramme> programmes = dataLayer.getProgrammes();
+    List<IProducer> producers = dataLayer.getProducers();
+    List<IPerson> person = dataLayer.getPersons();
+
+    @Override
+    public List<IProgramme> getProgrammes(int producerID) {
+        return dataLayer.getProducer(producerID).getProgrammes();
+    }
+
+    @Override
+    public boolean createProducer(String company, List <IProgramme> programmes) {
+        String user = "Admin";
+        dataLayer.createProducer(company, programmes);
+        dataLayer.logMessage(user + " added producer " + company);
+        return true;
+    }
+
+    //Get producerID from session
+    @Override
+    public boolean createProgramme(String name, Category category, String channel, Date airedDate) {
+        String user = (session.isAdmin) ? "Admin" : "Producer";
+        List<ICredit> credits = new ArrayList<>(0);
+        List<IProducer> producers = new ArrayList<>(0);
+        producers.add(dataLayer.getProducer(session.producerID));
+        dataLayer.createProgramme(name, category, channel, airedDate, credits, producers);
+        dataLayer.logMessage(user + " added programme: " + name);
+        return true;
+    }
+
+    @Override
+    public boolean createPerson(String name, Date birthdate, String description) {
+        String user = (session.isAdmin) ? "Admin" : "Producer";
+        dataLayer.createPerson(name, birthdate, description);
+        dataLayer.logMessage(user + " added person: " + name);
+        return true;
+    }
+
+    @Override
+    public boolean createCredit(IPerson person, FunctionType functionType) {
+        String user = (session.isAdmin) ? "Admin" : "Producer";
+        dataLayer.createCredit(person, functionType);
+        dataLayer.logMessage(user + " added credit for: " + person.getName() + " as " + functionType);
+        return true;
+    }
+
+    @Override
+    public boolean createCredit(String name, Date birthDate, String description, FunctionType functionType) {
+        String user = (session.isAdmin) ? "Admin" : "Producer";
+        int personID = dataLayer.createPerson(name, birthDate, description);
+        //createCredit returns int, maybe return -1 if credit couldn't be created
```

Comment:
The int returned is the id of the newly inserted credit.  And yes, `-1` when the credit couldn't be created.,